platform/android/audio_driver_opensl.cpp

```diff
@@ -211,7 +213,7 @@ Error AudioDriverOpenSL::init_input_device() {
 
 	SLDataLocator_AndroidSimpleBufferQueue loc_bq = {
 		SL_DATALOCATOR_ANDROIDSIMPLEBUFFERQUEUE,
-		2
+		1,
```

Comment:
What is this change for? I don't see it mentioned in the PR description, and seems unrelated without an explanation

Reply:
That's `SLDataLocator_AndroidSimpleBufferQueue::numBuffers` (https://android.googlesource.com/platform/frameworks/wilhelm/+/master/include/SLES/OpenSLES_Android.h#382), Godot uses just 1 buffer for reading input. It seems that this was copied from the output implementations, where 2 buffers are used. I don't think it actually impacts anything, as buffer queue isn't suppoerted for audio recorder on Android: https://developer.android.com/ndk/guides/audio/opensl/opensl-for-android#objects-and-interfaces.

Reply:
Kinda included it to make code more correct

Reply:
Official example seem to use one buffer: https://android.googlesource.com/platform/system/media/+/d5c746ba436dfc4605672f4a9f8c0628521a9f21/opensles/tests/examples/slesTestRecBuffQueue.cpp#179 and https://android.googlesource.com/platform/system/media/+/d5c746ba436dfc4605672f4a9f8c0628521a9f21/opensles/tests/examples/slesTestRecBuffQueue.cpp#34. Gonna be honest, I didn't know they existed. I'm gonna look through them and see if there are any other potential issues

Reply:
That sounds good, I just asked because you made no mention of this change in the PR description and it wasn't related to the fix itself so it's important to get a reason for the change

I'd personally suggest it should be done in a separate PR to make it easier to review and discuss and keep things focused on a specific fix

Reply:
> That sounds good, I just asked because you made no mention of this change in the PR description and it wasn't related to the fix itself so it's important to get a reason for the change

No worries :)

> I'd personally suggest it should be done in a separate PR to make it easier to review and discuss and keep things focused on a specific fix

Makes sense to me! #92983